### DISCUSSION TITLE
Source code for .so and .dll files

### DISCUSSION THREAD

**Discussion by rohinisingh-tech9** [llmware-ai/llmware#150] [Q&A] [answered]
llmware seems to be a great project to RAG and I am very interested in adopting llmware for a project I am working on . I am going through the Python code and can understand it.

However, please forgive my ignorance but I am not able to understand where can I get the code for .dll and .so files which are included in the /lib folder , especiall the ones which have _llmware as suffix ( e.g. .libpdf_llmware.so) . Before going full steam with using llmware I wasn to make sure that if need arises,  I should be able to  change the behaviour of those functionality which is provided by llmware binaries. Can anyone throw some light on if/where the code for those libraries are available and how those libraries are built ?

**Comment by doberst** [comment, marked as answer]
Thanks for your feedback!   We have not yet published the C source code for the PDF parser, Office parser and graph building methods.   It is intended to be fully part of this repository.   We are working on cleaning up the documentation and evaluating the best way to manage the lifecycle and ongoing pull requests on that source code.   We would be happy to discuss your project - and specific needs around the C source - and we can work with you to ensure that we address - either by adding to the current code - or providing some way to share with you so that you can build from source.

**Reply by rohinisingh-tech9** [reply]
Thank you very much for your kind answer. I am still working on something demonstrable and will surely bother you if I need any help or if I have some questions. Its early days with llmware but its something I was starting to build myself for my own use, surely not as general purpose as llmware. Will be in touch. Thanks again.
